src/docs/patterns/overlay.mdx

```diff
@@ -1,73 +1,132 @@
+<ImageGrid images={[Banner]} regularWidth />
 
-## Modal Dialog
+Overlays are useful in a variety of situations when you need to provide more context or information to a user than can be found in your primary content. For example, they can be used to present a confirmation dialog, details about a selected item, or actions that can be taken on a page. These can be used for presenting information (such as with a dialog) or capturing it (such as in a form).
+
+<Divider />
 
-The modal dialog is your typical pop-up window. Modals are great for presenting a single task to a user. By using a modal, you can allow the user to do more work without having to clutter your main layout with unnecessary information (e.g., an Add User button could open a modal with a form to enter the new user's information). These types of modals should be non-blocking, such that the user could click anywhere on the screen to dismiss it. Modals can also be used to edit large amounts of information that would be too much to fit into a sidebar.
+# Variations
 
-Modal dialogs are also very commonly used to confirm a user action (such as deleting an item). These usually appear in the center of the screen.
+Overlay elements come in a variety of shapes and sizes depending on their purpose and can be either blocking (i.e., the user must interact with the overlay before they can do anything else) or non-blocking.
+
+## Modal Dialog
 
-<MaterialDesignDescription
-    title={'Dialogs'}
-    url={'https://material.io/components/dialogs/'}
-    description={'Read more about dialogs on Material.io.'}
+<ImageGrid
+    images={[DesktopDialog, DesktopDialogTask]}
+    caption={
+        'A modal dialog appears centered in the screen over a dark overlay. These can present information to the user, or prompt them to enter information.'
+    }
 />
+<MaterialDesignDescription title={'Dialogs'} url={'https://material.io/components/dialogs/'} />
+
+The modal dialog is your typical pop-up window that appears in the center of the screen in front of primary content. The primary use of dialogs is to present critical information to users or to ask for a decision. For example, you may wish to ask a user if they are sure they want to delete a file.
+
+Modals are also great for presenting single tasks or workflows to a user. By using a modal, you allow the user to do more work without cluttering your main layout with unnecessary information (e.g., an Add User button could open a modal with a form to enter the new user's information). You can also use modals to edit large amounts of information that would be too difficult to fit into a sidebar.
+
+Modal dialogs are always blocking — they must be dismissed before the user can interact with the rest of the application content. They may be dismissed by clicking on a close button or, for less critical dialogs, by clicking anywhere on the screen outside of it.
 
```

Comment:
How should the modal dialog behave on a phone?

Reply:
When should I go for modal dialogs, and when should the application open a new page?

Reply:
If you tell me, I'll write it 😛 

Modals on the phone are the same as far as I am aware.

Reply:
Modals on the phone? 

![image](https://user-images.githubusercontent.com/8997218/93922162-92431d80-fcdf-11ea-9dd8-250a9a0ed88e.png)

Reply:
![image](https://user-images.githubusercontent.com/29152776/94000975-ded13c00-fd65-11ea-97e2-d727b151e929.png)

Reply:
I won't call that plagarism

Reply:
I used a similar one but flipped direction so I can talk about the desktop first